packages/asset-swapper/src/utils/quote_requestor.ts

```diff
@@ -27,6 +34,40 @@ function getTokenAddressOrThrow(assetData: string): string {
     throw new Error(`Decoded asset data (${JSON.stringify(decodedAssetData)}) does not contain a token address`);
 }
 
+function assertTakerAddressOrThrow(takerAddress: string | undefined): void {
+    if (
+        takerAddress === undefined ||
+        takerAddress === '' ||
+        takerAddress === '0x' ||
+        takerAddress === constants.NULL_ADDRESS
+    ) {
+        throw new Error('RFQ-T requires the presence of a taker address');
+    }
+}
+
+function inferQueryParams(
+    marketOperation: MarketOperation,
+    makerAssetData: string,
+    takerAssetData: string,
+    assetFillAmount: BigNumber,
+): { buyToken: string; sellToken: string; buyAmount: string | undefined; sellAmount: string | undefined } {
+    if (marketOperation === MarketOperation.Buy) {
+        return {
```

Comment:
Nit but it'd be nice to define the shared things at the top. 

Also I don't think we need the explicit `: undefined` definitions

```
const baseResult = { buyToken: getTokenAddressOrThrow(makerAssetData), sellToken:  getTokenAddressOrThrow(takerAssetData)};

if (marketOperation === MarketOperation.Buy) {
   return { ...baseResult, buyAmount: assetFillAmount.toString()};
else {
   return { ...baseResult, sellAmount: assetFillAmount.toString()};
}
```

Reply:
Please clarify what you'd like to see moved to the top.  Is it the return value type of this function?

Regarding the `undefined`s, `tsc` complains if I remove them:

```
$ /home/gene/dev/0x-monorepo/node_modules/.bin/tsc -b
src/utils/quote_requestor.ts:55:9 - error TS2322: Type '{ buyToken: string; sellToken: string; buyAmount: string; }' is not assignable to type '{ buyToken: string; sellToken: string; buyAmount: string | undefined; sellAmount: string | undefined; }'.
  Property 'sellAmount' is missing in type '{ buyToken: string; sellToken: string; buyAmount: string; }'.

 55         return {
            ~~~~~~~~
 56             buyToken: getTokenAddressOrThrow(makerAssetData),
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
... 
 58             buyAmount: assetFillAmount.toString(),
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 59         };
    ~~~~~~~~~~
```

Reply:
Ah sorry for not being clear here -- my suggestion was for the `buyToken` and `sellToken` assignments to be shared at the top of the function. 

Also, my bad on the `undefined` note.  Here's my suggested change:

```ts
const baseResult = { buyToken: getTokenAddressOrThrow(makerAssetData), sellToken:  getTokenAddressOrThrow(takerAssetData)};

if (marketOperation === MarketOperation.Buy) {
   return { ...baseResult, buyAmount: assetFillAmount.toString(), sellAmount: undefined };
else {
   return { ...baseResult, buyAmount: undefined, sellAmount: assetFillAmount.toString()};
}
```

This is def a nit tho -- totally OK with me if you want to keep as-is